cf-agent/verify_files.c

```diff
@@ -804,7 +803,18 @@ static PromiseResult RenderTemplateMustache(EvalContext *ctx, const Promise *pp,
                               "update rendering of '%s' from mustache template '%s'",
                               edcontext->filename, message))
             {
-                if (SaveAsFile(SaveBufferCallback, output_buffer, edcontext->changes_filename, attr, edcontext->new_line_mode))
+                if (!fileexists && !CfCreateFile(ctx,
+                                                 edcontext->changes_filename,
+                                                 pp, attr, &result))
```

Comment:
So the `create` attribute is ignored?

I think we either have to respect `create => "false"` or make that invalid. I suggest adding an error and promise failed if you both promise template editing and `create => "false"`. There is a much better way to express that something should happen only if the file already exists (`if => fileexists()`). Also, could you check how this works in `content` attribute?

Reply:
Looks like the `content` attribute ignores `create => "false"` as well.

```
$ cat ~/test.cf
bundle agent __main__
{
  files:
      "/tmp/test_1"
        content => "Hello World!";
      "/tmp/test_2"
        create => "true",
        content => "Hello World!";
      "/tmp/test_3"
        create => "false",
        content => "Hello World!";
}

$ rm -f /tmp/test_?
$ cf-agent  -KIf ~/test.cf 
    info: Updated content of '/tmp/test_1' with content 'Hello World!'
    info: Created file '/tmp/test_2', mode 0600
    info: Updated content of '/tmp/test_2' with content 'Hello World!'
    info: Updated content of '/tmp/test_3' with content 'Hello World!'
$ ls /tmp/test_?
/tmp/test_1  /tmp/test_2  /tmp/test_3
```